### PR TITLE
Wip client misc

### PR DESCRIPTION
**Pull request by ukernel**

### CODE REVIEW

**Comment by jcsp** [comment]
Both these changes look :+1: to me.  The cache invalidation stuff has been twitchy in the past, so let's double review this: @gregsfortytwo

File: src/client/Client.cc [hunk 1]
```diff
@@ -10870,3 +10880,23 @@ void Client::set_cap_epoch_barrier(epoch_t e)
...
+void Client::handle_conf_change(const struct md_config_t *conf,
+				const std::set <std::string> &changed)
+{
+  if (changed.count("client_cache_size") ||
+      changed.count("client_cache_mid")) {
+    lru.lru_set_max(cct->_conf->client_cache_size);
+    lru.lru_set_midpoint(cct->_conf->client_cache_mid);
+  }
+}
+
```
Comment (gregsfortytwo) [inline src/client/Client.cc]: This is a good idea, but you also need to call g_conf->add_observer with the Client to register for updates. ;)

**Comment by gregsfortytwo** [comment]
Please write a ceph-qa-suite test case demonstrating that this works. What I'd really like is one that sets very low cache sizes on the MDS, disables kernel invalidation, makes sure we get a warning about cache pressure, and then enables it and shows that it goes away. :)

File: src/client/Client.cc [hunk 2]
```diff
@@ -431,6 +431,8 @@ int Client::init()
 
   client_lock.Unlock();
 
+  g_conf->add_observer(this);
```
Comment (gregsfortytwo) [inline src/client/Client.cc]: Heh, this should be cct, not g_conf — build failure! Guess my advice was a little slapdash. ;)

**Comment by ukernel** [comment]
https://github.com/ceph/ceph-qa-suite/pull/442

**Comment by gregsfortytwo** [comment]
Running through greg-fs-testing.

**Pull request merged**